src/NovelRT/Ecs/SystemScheduler.cpp

```diff
@@ -157,6 +177,30 @@ namespace NovelRT::Ecs
 
                     pair.threadLock.unlock();
                 }
+
+                if (remainder % amountOfWork != 0)
+                {
+                    size_t threadWorkIndex = ((remainder / amountOfWork) + 1) < _threadWorkQueues.size()
+                                                 ? ((remainder / amountOfWork) + 1)
+                                                 : 0;
+
+                    QueueLockPair& pair = _threadWorkQueues[threadWorkIndex];
+                    size_t startingIndex = _systemIds.size() - sizeOfProcessedWork;
+
+                    pair.threadLock.lock();
+
+                    if ((_threadAvailabilityMap & 1ULL << threadWorkIndex) == 0)
+                    {
+                        _threadAvailabilityMap ^= 1ULL << threadWorkIndex;
+                    }
+
+                    for (size_t i = startingIndex; i < _systemIds.size(); i++)
+                    {
+                        pair.systemUpdateIds.push_back(_systemIds[i]);
+                    }
+
+                    pair.threadLock.unlock();
+                }
```

Comment:
I'm not sure I entirely understand this block. Could you clarify?

Reply:
Yea! While thinking about possible situations I realized that sometimes the else statement could drop systems.

Take for example 7 workers and 19 systems, 'amountOfWork' in this case = 2, 'remainder' = 5.

In the else statement the outer for statement will iterate over 0 <= i < 2 (5/2), inner for iterates 0 <= i < 2 (amount of work). This is 4 total operations. Which means the last one is dropped.

My 'if (remainder % amountOfWork != 0)' checks for this situation.

-------------------------------------------------------------------------------------
As for how it works, 'threadWorkIndex' is set to the next worker, '(remainder / amountOfWork)' being the previous upper bound for the outer for statement from before, which becomes the index of the worker that takes care of these extra systems. (realized I didnt need the +1 in this explanation)

The reason I'm using a ternary operator is to make sure that the selected thread exists. I'm not sure if it is needed, but working with you yesterday made it seem like you felt that the remainder could be > #ofworkers, so I added that check.

.size() - sizeOfProcessedWork should equal the starting index of the remainder of the remainder, and then those left over systems are assigned!

Reply:
have you got a unit test to confirm this scenario?

Reply:
Yes, the last Unit test I added should hit it on most computers.
'IndependentSystemsCanHandleManySystems' the way it works is iterates 4 times using 11, 17, 23, and 37 systems. Prime numbers so that they dont divide evenly. I felt those four will be good, but let me know if I should add more.

Reply:
Okay, sounds good enough for me. Good work mate. 🎉